ofrak_core/test_ofrak/unit/test_resource.py

```diff
@@ -170,3 +179,37 @@ async def test_get_most_specific_tags(resource: Resource):
 
     expected_most_specific_tags = {GenericText, FilesystemRoot, Elf}
     assert set(resource.get_most_specific_tags()) == expected_most_specific_tags
+
+
+@pytest.fixture(autouse=True)
+def mock_ofrak_component(ofrak):
+    ofrak.injector.discover(mock_component)
+
+
+@pytest.mark.asyncio
+async def test_flush_to_disk_pack(ofrak_context: OFRAKContext):
+    # This test works as long as LZMA fails to pack() and the default is to
+    # pack recursively. The root resource is a LZMA archive in a LZMA archive,
+    # which as of 2022-09-13 will fail to pack and therefore pack_recursively
+    # won't work.
+
+    root_resource = await ofrak_context.create_root_resource("mock", b"\x00" * 0x100)
+    root_resource.add_tag(MockFile)
+
+    child = await root_resource.create_child((MockFailFile,), data_range=Range(0x10, 0x20))
+    child.add_tag(MockFailFile)
+
+    with BytesIO() as buffer:
+        # should fail because write_to runs pack_recursively and MockFailFile will fail on packing
+        with pytest.raises(ComponentAutoRunFailure):
```

Comment:
Let me add a proper [Exception catch](https://docs.python.org/3.7/library/exceptions.html#built-in-exceptions
The expression following from must be an exception or None. It will be set as __cause__ on the raised exception.)

Reply:
This the only thing blocking this MR for me.